### ISSUE TITLE
Search function doesn't seem to actually check if string is *contained* within field

### ISSUE THREAD

**Issue by leosco** [mscdex/node-imap#647]
I'm trying to search for messages that *contain* the subject header 'node-imap', just as displayed in the example in the docs. I emphasize *contain* because my use case for the search function would be to return all emails that have 'node-imap' in the subject, i.e. 'node-imap-1', 'I love node-imap', etc.

However, it appears to only return the messages where the subject is exactly *equal* to 'node-imap', not necessarily if 'node-imap' is contained within the string. The exact language given in the example is 
"All messages that have 'node-imap' in the subject header: [ ['HEADER', 'SUBJECT', 'node-imap'] ]"

This is the code I'm using: 
`imap.search([['HEADER', 'SUBJECT', 'node-imap']], (err, uids) => console.log(uids))`.
And when called, it only returns the uid of the one message whose subject is exactly 'node-imap' not any of the other ones, where 'node-imap' is contained in the subject.

Any thoughts on this?

**Comment by mscdex** [comment]
This is a limitation of IMAP itself as header and subject searches are always substring searches. There is nothing that can be done about that, unless the server offers a custom extension that provides support for such searching (I don't think even gmail's `X-GM-RAW` extension allows this).

**Comment by leosco** [comment]
Thanks for the quick response! But yeah, I tried both, and neither of them work. It's pretty annoying that the [IMAP RFC](https://tools.ietf.org/html/rfc3501#page-49) explicitly states:
```
      In all search keys that use strings, a message matches the key if
      the string is a substring of the field.  The matching is
      case-insensitive.
```
And it just doesn't work 😢😢😢

**Comment by leosco** [comment]
https://stackoverflow.com/questions/13403790/python-imap-search-for-partial-subject

So according to that, Google doesn't implement IMAP properly, and only matches partial criteria by whole words, not substring. Therefore, from a search for 'node-imap' in
```
node-imap
node-imap2
I love node-imap
hello-node-imap
```
it would only return 'node-imap' and 'I love node-imap', since the search term is present as a complete word. It's super weird that they'd do it this way. 

I guess I'll do more research on it, maybe I can figure it out, at least for Gmail.

**Comment by mscdex** [comment]
Yeah it looks like you might have to find some other workaround. Perhaps use a different filter to get a smaller subset of the messages and then retrieve just the subject of each message and do the subject filtering in your code.

**Comment by leosco** [comment]
I was able to tweak my subject lines to accommodate the whole "search by word, not substring" business, but either way I don't think I'm going to be able to use this package for this particular feature I'm working on, since I'm dealing with large attachments. Gmail's limit on attachment size is 23 MB and Google Drive API's limit on file exports is only 10 MB, annoyingly. 

So I've decided to use Dropbox instead for pulling large files into Node. I'm also closing this issue since the root of the problem was a Gmail limitation instead of anything wrong with node-imap.

Thanks a bunch for your help @mscdex !
